### ISSUE TITLE
[ENHANCEMENT] let user could setup `HostStorageClass` with helm option

### ISSUE THREAD

**Issue by Vicente-Cheng** [harvester/harvester#4045]
**Is your enhancement related to a problem? Please describe.**
The environment variable `HostStorageClass` could be set for the default StorageClass associated with the bare-metal harvester cluster of the harvester CSI driver.
We may expose this variable to set it up when deploying the downstream cluster.

**Describe the solution you'd like**
Expose the `HostStorageClass` with the helm chart

**Describe alternatives you've considered**
None

**Additional context**
Reference for this variable:
https://github.com/harvester/harvester-csi-driver/blob/master/main.go#L41

**Comment by irishgordo** [comment]
Hi, @Vicente-Cheng - I've been working through testing this but had some validation questions.

Is it possible to validate this by "re-deploying" the harvester-csi-driver App from Rancher UI w/ "Edit YAML" and on the `helm options`:
- Cleanup on Failure
- Execute chart hooks
- Force
- Reset Values
- Wait
( all boxes checked )
to update it?

As I've been trying with separate loadouts of "Edit YAML" either taking the shape of the following.

Edited YAML Example A:
```yaml
cloudConfig:
  hostPath: /var/lib/rancher/rke2/etc/config-files/
  secretName: ''
fullnameOverride: ''
global:
  cattle:
    systemProjectId: p-cgbv4
  clusterCIDR: 10.42.0.0/16
  clusterCIDRv4: 10.42.0.0/16
  clusterDNS: 10.43.0.10
  clusterDomain: cluster.local
  rke2DataDir: /var/lib/rancher/rke2
  serviceCIDR: 10.43.0.0/16
image:
  csi:
    attacher:
      repository: rancher/mirrored-longhornio-csi-attacher
      tag: v3.2.1
    nodeDriverRegistrar:
      repository: rancher/mirrored-longhornio-csi-node-driver-registrar
      tag: v2.3.0
    provisioner:
      repository: rancher/mirrored-longhornio-csi-provisioner
      tag: v2.1.2
    resizer:
      repository: rancher/mirrored-longhornio-csi-resizer
      tag: v1.2.0
  harvester:
    csiDriver:
      repository: rancher/harvester-csi-driver
      tag: v0.1.5
      args:
        - '--host-storage-class=extrasc'
  pullPolicy: IfNotPresent
kubeletRootDir: /var/lib/kubelet
nameOverride: ''
nodeSelector:
  kubernetes.io/os: linux
replicasCount: 3
tolerations:
  - effect: NoSchedule
    key: kubevirt.io/drain
    operator: Exists
  - effect: NoSchedule
    key: node-role.kubernetes.io/control-plane
    operator: Equal
  - effect: NoExecute
    key: node-role.kubernetes.io/etcd
    operator: Equal
  - effect: NoSchedule
    key: cattle.io/os
    operator: Equal
    value: linux
```

Edited YAML Example B:
```yaml
cloudConfig:
  hostPath: /var/lib/rancher/rke2/etc/config-files/
  secretName: ''
fullnameOverride: ''
global:
  cattle:
    systemProjectId: p-cgbv4
  clusterCIDR: 10.42.0.0/16
  clusterCIDRv4: 10.42.0.0/16
  clusterDNS: 10.43.0.10
  clusterDomain: cluster.local
  rke2DataDir: /var/lib/rancher/rke2
  serviceCIDR: 10.43.0.0/16
image:
  csi:
    attacher:
      repository: rancher/mirrored-longhornio-csi-attacher
      tag: v3.2.1
    nodeDriverRegistrar:
      repository: rancher/mirrored-longhornio-csi-node-driver-registrar
      tag: v2.3.0
    provisioner:
      repository: rancher/mirrored-longhornio-csi-provisioner
      tag: v2.1.2
    resizer:
      repository: rancher/mirrored-longhornio-csi-resizer
      tag: v1.2.0
  harvester:
    csiDriver:
      repository: rancher/harvester-csi-driver
      tag: v0.1.5
      env:
        - name: HOST_STORAGE_CLASS
          value: extrasc
  pullPolicy: IfNotPresent
kubeletRootDir: /var/lib/kubelet
nameOverride: ''
nodeSelector:
  kubernetes.io/os: linux
replicasCount: 3
tolerations:
  - effect: NoSchedule
    key: kubevirt.io/drain
    operator: Exists
  - effect: NoSchedule
    key: node-role.kubernetes.io/control-plane
    operator: Equal
  - effect: NoExecute
    key: node-role.kubernetes.io/etcd
    operator: Equal
  - effect: NoSchedule
    key: cattle.io/os
    operator: Equal
    value: linux
```

With either edited yaml loadouts on:
- Harvester v1.3.0-rc2
- Rancher v2.8.1 (on rancher-vcluster addon)

I'm not able to build a PVC for the RKE2 (Kubernetes Version: v1.27.10 +rke2r1) cluster that when selecting `harvester` will utilize the `extrasc` storage class, it's still all defaulting to the default storage-class.

Artifacts:
![Screenshot from 2024-02-13 16-00-48](https://github.com/harvester/harvester/assets/5370752/9ae87b53-d6df-42f5-9614-aeb1aefec958)
![Screenshot from 2024-02-13 15-59-58](https://github.com/harvester/harvester/assets/5370752/19ffd49f-aa9e-4f46-87b0-284dd760873a)
![Screenshot from 2024-02-13 15-56-06](https://github.com/harvester/harvester/assets/5370752/43ba925c-f145-4cdc-9596-f31c5a9bf2dd)
![Screenshot from 2024-02-13 15-44-43](https://github.com/harvester/harvester/assets/5370752/ece118c5-487c-4e8a-b21c-e40732535bfc)

**Comment by Vicente-Cheng** [comment]
Hi @irishgordo,
I thought I needed some clarification here.

I move this one back to the `Design` to clarify again.
Thanks for the testing!

**Comment by Vicente-Cheng** [comment]
Update the test plan. now we can use helm option to setup HostStorageClass

**Comment by irishgordo** [comment]
@Vicente-Cheng thanks for updating the test plan :smile: :+1: 
**Vallidation Question:**
- I was wondering, should I be seeing the CSI App as a Chart listed in All Catagories from your rancher-charts repo?
![Screenshot from 2024-10-23 17-43-49](https://github.com/user-attachments/assets/2822303a-28f0-4894-8482-6a5b2a484d01)
![Screenshot from 2024-10-23 17-43-12](https://github.com/user-attachments/assets/a7a19634-a51a-4600-8479-1eefc8c1baaf)

Or is that not needed to do the manual chart upgrade?

**Comment by Vicente-Cheng** [comment]
The PR is already merged into dev-v2.8 branch on https://github.com/rancher/charts.git

So, you can try to use the repo: https://github.com/rancher/charts.git, branch: dev-v2.8
You can search harvester-csi (by filter) on the Apps->Charts
It should have an extra one

**Comment by irishgordo** [comment]
Thanks for that mention @Vicente-Cheng 
This all looks great on v1.4.0-rc4 w/ Rancher v2.9.2 (vCluster, k3s v1.29)
Upgrade of CSI 0.1.18 -> 0.1.20 w/ hostStorageClass

Supporting Screengrabs:
![Screenshot from 2024-10-23 22-48-18](https://github.com/user-attachments/assets/c8086cb4-4126-4a14-b501-777aa5ed3bb8)
![Screenshot from 2024-10-23 22-47-59](https://github.com/user-attachments/assets/04cece8f-8d28-4960-bc24-fdbe46a47c32)
![Screenshot from 2024-10-23 22-47-02](https://github.com/user-attachments/assets/cda6e573-1201-4bce-a960-9a13d39aabd7)
![Screenshot from 2024-10-23 22-44-04](https://github.com/user-attachments/assets/80fa1316-61ef-4638-bfea-b5d93a773957)
![Screenshot from 2024-10-23 22-43-30](https://github.com/user-attachments/assets/8ce2884c-b329-4473-96df-fede9b89e8ae)
![Screenshot from 2024-10-23 22-42-37](https://github.com/user-attachments/assets/e864e68e-2b4d-472c-b28a-ad467614eca1)
![Screenshot from 2024-10-23 22-41-27](https://github.com/user-attachments/assets/99fbd2f0-7279-4d43-852c-614668c83d28)
![Screenshot from 2024-10-23 22-40-58](https://github.com/user-attachments/assets/0f0ec709-73a7-4ec4-9c3e-4653a5205fe2)
![Screenshot from 2024-10-23 22-40-12](https://github.com/user-attachments/assets/b235e407-2fec-4e30-89e6-2e4c6fab9f7f)
![Screenshot from 2024-10-23 22-37-43](https://github.com/user-attachments/assets/afda4b90-498e-477d-8c17-4dc6264cf30d)
![Screenshot from 2024-10-23 22-35-46](https://github.com/user-attachments/assets/46b60447-3477-427c-941e-a888d05e5158)

This all looks good :smile: :+1:  -> I'll go ahead and close this out :+1:
